src/join.t.sol

```diff
@@ -861,4 +932,26 @@ contract DssDeployTest is DssDeployTestBase {
         saiJoin.deny(address(this));
         saiJoin.join(address(this), 10);
     }
+
+    function testTokenManaged() public {
+        deployKeepAuth();
+        DSValue pip = new DSValue();
+
+        DSToken wbtc = new DSToken("WBTC");
+        wbtc.mint(10);
+        ManagedGemJoin wbtcJoin = new ManagedGemJoin(address(vat), "WBTC", address(wbtc));
+        assertEq(wbtcJoin.dec(), 18);
+
+        dssDeploy.deployCollateral("WBTC", address(wbtcJoin), address(pip));
+
+        wbtc.approve(address(wbtcJoin), uint256(-1));
+        assertEq(wbtc.balanceOf(address(wbtcJoin)), 0);
+        assertEq(vat.gem("WBTC", address(this)), 0);
+        wbtcJoin.join(address(this), 10);
+        assertEq(wbtc.balanceOf(address(wbtcJoin)), 10);
+        assertEq(vat.gem("WBTC", address(this)), 10);
+        wbtcJoin.exit(address(this), address(this), 4);
+        assertEq(wbtc.balanceOf(address(wbtcJoin)), 6);
+        assertEq(vat.gem("WBTC", address(this)), 6);
```

Comment:
For completeness should also check that `wbtc.balanceOf(address(this)) == 4`.

Reply:
Done